sim/priest/shadow/rotation.go

```diff
@@ -68,10 +69,15 @@ func (spriest *ShadowPriest) tryUseGCD(sim *core.Simulation) {
 	// initialize helpful variables for calculations later
 	vtCastTime := spriest.ApplyCastSpeed(time.Millisecond * 1500)
 	gcd := spriest.SpellGCD()
-	tickLength := spriest.ApplyCastSpeed(time.Second)
+	var mf_reduc_time time.Duration
+	if spriest.HasSetBonus(priest.ItemSetCrimsonAcolyte, 4) {
+		mf_reduc_time = time.Millisecond * 510
+	}
+	tickLength := spriest.ApplyCastSpeed(time.Second) - mf_reduc_time
 	DotTickSpeed := float64(spriest.ApplyCastSpeed(time.Second * 3))
 	critChance := (spriest.GetStat(stats.SpellCrit) / (core.CritRatingPerCritChance * 100))
 	remain_fight := float64(sim.GetRemainingDuration())
+	bosshealth := float64(sim.GetRemainingDurationPercent())
```

Comment:
We have a built-in mechanism for this, which you need to use in order to work properly with the user inputs in the UI. Call this function to check for 35% execute: https://github.com/wowsims/wotlk/blob/master/sim/core/sim.go#L363

Reply:
updated